superset/assets/src/dashboard/util/publishSubscriberUtil.js

```diff
@@ -30,17 +32,62 @@ export function getUniqueActionsForSlice(slice) {
 export function getModalSliceIDFor(publishSubscriberMap, publisherId) {
     if (publishSubscriberMap && publishSubscriberMap.hasOwnProperty('subscribers')) {
         const subscribers = Object.values(publishSubscriberMap.subscribers);
-        var item = _.find(subscribers, function (item) {
+        var item = find(subscribers, function (item) {
             return ((item.actions.indexOf(USE_AS_MODAL) > -1) && (Object.keys(item.linked_slices).indexOf(publisherId) != -1));
         });
         return item ? item.id : undefined;
     }
     return undefined
 }
 
+export function getSlicesWithSubHeader(subscribers, chartId, filters) {
+    let subHeader = '';
+    if (chartId != -1 && subscribers && subscribers[chartId]) {
+        const subscriber = subscribers[chartId];
+        if (subscriber.actions.indexOf(INCLUDE_IN_TITLE) > -1) {
+            subHeader = '';
+            for (let lsKey in subscriber.linked_slices) {
+                if (keyExists(lsKey, filters)) {
+                    let linkedSlicesLength = subscriber.linked_slices[lsKey].length;
+
+                    for (let i = 0; i < linkedSlicesLength; i++) {
+                        let linkedSlice = subscriber.linked_slices[lsKey];
+
+                        if (linkedSlice[i].actions.indexOf(INCLUDE_IN_TITLE) > -1) {
+                            let columnName = linkedSlice[i]['col'];
+                            let subHeaderValues = filters[lsKey][columnName];
+                            const values = subHeaderValues ? subHeaderValues.constructor == Array ? subHeaderValues.concat() : [subHeaderValues] : [];
+
+                            if (subHeaderValues) {
+                                (subHeader != '') ? values.push(subHeader) : [];
+                                subHeader = values.join(' | ');
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    return subHeader;
+}
+
+export function keyExists(key, search) {
+    if (!search || (search.constructor !== Array && search.constructor !== Object)) {
+        return false;
+    }
+    for (let i = 0; i < search.length; i++) {
+        if (search[i] === key) {
+            return true;
+        }
+    }
```

Comment:
we can use lodsh find   and  indexOf method instead loop

Reply:
If you see the method `keyExists`, it's a combination of native javascript Array and constructor, you can use this method commonly across everywhere instead of `lodash`.